app-backend/batch/src/main/scala/stacExport/LabelCollectionBuilder.scala

```diff
@@ -0,0 +1,299 @@
+    new LabelCollectionBuilder(
+      labelCollection.copy(stacVersion = Some(version))
+    )
+
+  def withId(
+      id: String
+  ): LabelCollectionBuilder[CollectionRequirements with CollectionId] =
+    new LabelCollectionBuilder(labelCollection.copy(id = Some(id)))
+
+  def withTitle(
+      title: String
+  ): LabelCollectionBuilder[CollectionRequirements with CollectionTitle] =
+    new LabelCollectionBuilder(labelCollection.copy(title = Some(title)))
+
+  def withDescription(
+      description: String
+  ): LabelCollectionBuilder[CollectionRequirements with CollectionDescription] =
+    new LabelCollectionBuilder(
+      labelCollection.copy(description = Some(description))
+    )
+
+  def withLinks(
+      links: List[StacLink]
+  ): LabelCollectionBuilder[CollectionRequirements with CollectionLinks] =
+    new LabelCollectionBuilder(
+      labelCollection.copy(links = labelCollection.links ++ links)
+    )
+
+  def withExtent(
+      extent: StacExtent
+  ): LabelCollectionBuilder[CollectionRequirements with CollectionExtent] =
+    new LabelCollectionBuilder(labelCollection.copy(extent = Some(extent)))
+
+  def withParentPath(
+      parentPath: String,
+      rootPath: String
+  ): LabelCollectionBuilder[CollectionRequirements with CollectionParentPath] =
+    new LabelCollectionBuilder(
+      labelCollection
+        .copy(parentPath = Some(parentPath), rootPath = Some(rootPath))
+    )
+
+  def withTasks(
+      tasks: List[Task]
+  ): LabelCollectionBuilder[CollectionRequirements with CollectionTasks] =
+    new LabelCollectionBuilder(
+      labelCollection.copy(tasks = labelCollection.tasks ++ tasks)
+    )
+
+  def withItemPropInfo(
+      itemPropsThin: StacLabelItemPropertiesThin
+  ): LabelCollectionBuilder[
+    CollectionRequirements with CollectionItemInfoThin
+  ] =
+    new LabelCollectionBuilder(
+      labelCollection.copy(itemPropsThin = itemPropsThin)
+    )
+
+  def withSceneItemLinks(
+      sceneItemLinks: List[(String, String)]
+  ): LabelCollectionBuilder[
+    CollectionRequirements with CollectionSceneItemLinks
+  ] =
+    new LabelCollectionBuilder(
+      labelCollection.copy(
+        sceneItemLinks = labelCollection.sceneItemLinks ++ sceneItemLinks
+      )
+    )
+
+  @SuppressWarnings(Array("OptionGet"))
+  def build()(
+      implicit ev: CollectionRequirements =:= CompleteCollection
+  ): (StacCollection, StacItem, String) = {
+    // Silence unused warning because scalac warns about phantom types
+    ev.unused
+    // s3://rasterfoundry-production-data-us-east-1/stac-exports/<catalogId>/<layerCollectionId>/<labelCollectionId>
+    val absPath = labelCollection.parentPath.get
+    // ../../../catalog.json
+    val rootPath = labelCollection.rootPath.get
+
+    val itemBuilder =
+      new StacItemBuilder[StacItemBuilder.ItemBuilder.EmptyItem]()
+    val labelItemId = UUID.randomUUID().toString()
+    val labelItemFootprint: MultiPolygon = labelCollection.tasks
```

Comment:
I am now thinking about moving this part to the `TaskDao` and let PostGIS do the union, since, as I experimented yesterday, it was really fast in PostGIS.